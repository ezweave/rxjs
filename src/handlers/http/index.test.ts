import { getCraftAndCrew } from '.';

describe(getCraftAndCrew, () => {
  it('parallelizes all the calls', async () => {
    const result = await getCraftAndCrew();
    console.warn('RESULT', result);
    expect(result).toMatchSnapshot();
  });
});
