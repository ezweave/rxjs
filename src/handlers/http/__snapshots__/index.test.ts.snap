// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCraftAndCrew parallelizes all the calls 1`] = `
Object {
  "MGLT": "75",
  "cargo_capacity": "100000",
  "consumables": "2 months",
  "cost_in_credits": "100000",
  "created": "2014-12-10T16:59:45.094000Z",
  "crew": "4",
  "edited": "2014-12-20T21:23:49.880000Z",
  "films": Array [
    "https://swapi.dev/api/films/1/",
    "https://swapi.dev/api/films/2/",
    "https://swapi.dev/api/films/3/",
  ],
  "hyperdrive_rating": "0.5",
  "length": "34.37",
  "manufacturer": "Corellian Engineering Corporation",
  "max_atmosphering_speed": "1050",
  "model": "YT-1300 light freighter",
  "name": "Millennium Falcon",
  "passengers": Array [
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-09T13:50:51.644000Z",
      "edited": "2014-12-20T21:17:56.891000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "male",
      "hair_color": "blond",
      "height": "172",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "77",
      "name": "Luke Skywalker",
      "skin_color": "fair",
      "species": Array [],
      "starships": Array [
        "https://swapi.dev/api/starships/12/",
        "https://swapi.dev/api/starships/22/",
      ],
      "url": "https://swapi.dev/api/people/1/",
      "vehicles": Array [
        "https://swapi.dev/api/vehicles/14/",
        "https://swapi.dev/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "112BBY",
      "created": "2014-12-10T15:10:51.357000Z",
      "edited": "2014-12-20T21:17:50.309000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/4/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "167",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "75",
      "name": "C-3PO",
      "skin_color": "gold",
      "species": Array [
        "https://swapi.dev/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/2/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "33BBY",
      "created": "2014-12-10T15:11:50.376000Z",
      "edited": "2014-12-20T21:17:50.311000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/4/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "96",
      "homeworld": "https://swapi.dev/api/planets/8/",
      "mass": "32",
      "name": "R2-D2",
      "skin_color": "white, blue",
      "species": Array [
        "https://swapi.dev/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/3/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "41.9BBY",
      "created": "2014-12-10T15:18:20.704000Z",
      "edited": "2014-12-20T21:17:50.313000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "male",
      "hair_color": "none",
      "height": "202",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "136",
      "name": "Darth Vader",
      "skin_color": "white",
      "species": Array [],
      "starships": Array [
        "https://swapi.dev/api/starships/13/",
      ],
      "url": "https://swapi.dev/api/people/4/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "150",
      "homeworld": "https://swapi.dev/api/planets/2/",
      "mass": "49",
      "name": "Leia Organa",
      "skin_color": "light",
      "species": Array [],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/5/",
      "vehicles": Array [
        "https://swapi.dev/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "52BBY",
      "created": "2014-12-10T15:52:14.024000Z",
      "edited": "2014-12-20T21:17:50.317000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "male",
      "hair_color": "brown, grey",
      "height": "178",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "120",
      "name": "Owen Lars",
      "skin_color": "light",
      "species": Array [],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/6/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "47BBY",
      "created": "2014-12-10T15:53:41.121000Z",
      "edited": "2014-12-20T21:17:50.319000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/5/",
        "https://swapi.dev/api/films/6/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "165",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "75",
      "name": "Beru Whitesun lars",
      "skin_color": "light",
      "species": Array [],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/7/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-10T15:57:50.959000Z",
      "edited": "2014-12-20T21:17:50.321000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.dev/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "97",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "32",
      "name": "R5-D4",
      "skin_color": "white, red",
      "species": Array [
        "https://swapi.dev/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.dev/api/people/8/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "24BBY",
      "created": "2014-12-10T15:59:50.509000Z",
      "edited": "2014-12-20T21:17:50.323000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.dev/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "black",
      "height": "183",
      "homeworld": "https://swapi.dev/api/planets/1/",
      "mass": "84",
      "name": "Biggs Darklighter",
      "skin_color": "light",
      "species": Array [],
      "starships": Array [
        "https://swapi.dev/api/starships/12/",
      ],
      "url": "https://swapi.dev/api/people/9/",
      "vehicles": Array [],
    },
  ],
  "pilots": Array [
    "https://swapi.dev/api/people/13/",
    "https://swapi.dev/api/people/14/",
    "https://swapi.dev/api/people/25/",
    "https://swapi.dev/api/people/31/",
  ],
  "starship_class": "Light freighter",
  "url": "https://swapi.dev/api/starships/10/",
}
`;
